ADDING_NEW_CLIENTS.md

```diff
@@ -0,0 +1,108 @@
+
+In general, make classes immutable whenever possible, providing builders as necessary.  Make model object classes `java.util.Serializable`.  Also, make classes final when possible, except when the class contains functionality that cannot be fully tested by users without mocking the object.  If a class cannot be made final, then `hashCode` or `equals` overrides should be made final if possible.
+
+`gcloud-java-core` provides functionality for code patterns used across `gcloud-java` libraries.  The following are some important core concepts:
+* Paging: Cloud services often expose page-based listing using page tokens.  The [`Page`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/cloud/Page.java) interface should be used for page-based listing.  A `Page` contains an iterator over results in that page, as well as methods to get the next page and all results in future pages.  `Page` requires a `NextPageFetcher` implementation (see the `NextPageFetcher` interface in [`PageImpl`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/cloud/PageImpl.java)).  This implementation should delegate constructing request options to the `nextRequestOptions` method in `PageImpl`.
+
+* Exception handling: we have the notion of retryable vs non-retryable exceptions and idempotent vs non-idempotent exceptions.  These are encapsulated by `BaseServiceException`.  Retryable error codes should be listed in the client's subclass of `BaseServiceException`, specifying whether exception is idempotent.  The subclass should also provide methods to translate the exceptions given by the underlying autogeneraged client library.  The [`ExceptionHandler`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/cloud/ExceptionHandler.java) class intercepts and decides how to handle exceptions (based on the set of retryable error codes) when making RPC calls.
+
+* Batching: The [`BatchResult`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/dns-alpha-batch/gcloud-java-core/src/main/java/com/google/cloud/BatchResult.java) class provides a simple way for users to combine RPC calls for performance enhancement.  Clients for services that provide batching should provide a batch class that contains methods similar to the methods in the `Service.java` subclass, but instead return a subclass of `BatchResult`.  Also provide an SPI-layer class to collect batch requests and submit the batch.
+
+Notes/reminders:
+* API layer classes should be located in the package `com.google.cloud.servicename`, where "servicename" corresponds to the name of the Cloud service.
+* Override the `ServiceOptions.defaultRetryParams()` method in your service's options class to align with the Service Level Agreement (SLA) given by the underlying service.  See #857 and #860 for context.
```

Comment:
Yeah, I think that's because this README is in my fork rather than merged into master.  Once it's merged, it should work.